lib/HALTest/IMU.h

```diff
@@ -0,0 +1,146 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  IMU realization
+ * @author Juliane Kerpe <juliane.kerpe@web.de>
+ *
+ * @addtogroup HALSim
+ *
+ * @{
+ */
+
+#ifndef IMU_H
+#define IMU_H
+
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "IIMU.h"
+
+class IMU : public IIMU
+{
+public:
+    /**
+     * Destroys the interface.
+     */
+    ~IMU()
+    {
+    }
+
+    /**
+     * Initializes the inertial sensors and detects their type.
+     *
+     * @return True if the sensor type was detected succesfully; false otherwise.
+     */
+    bool init(){return true;}
+        
+    /**
+     * Enables all of the inertial sensors with a default configuration.
+     */
+    void enableDefault() {};
+
+    /**
+     * 	Configures the sensors with settings optimized for turn sensing.
+     */
+    void configureForTurnSensing() {}
+
+    /**
+     * Takes a reading from the accelerometer and makes the measurements available in a.
+     */ 
+    void readAccelerometer() {}
+        
+    /**
+     * Takes a reading from the gyro and makes the measurements available in g.
+     */
+    void readGyro() {}
+        
+    /**
+     * Takes a reading from the magnetometer and makes the measurements available in m.
+     */
+    void readMagnetometer() {}
+        
+    /**
+     * Indicates whether the accelerometer has new measurement data ready.
+     * 
+     * @return True if there is new accelerometer data available; false otherwise. 
+     */
+    bool accelerometerDataReady() {return true;}
+        
+    /**
+     * Indicates whether the gyro  has new measurement data ready.
+     * 
+     * @return True if there is new gyro data available; false otherwise.
+     */
+    bool gyroDataReady() {return true;}
+        
+    /**
+     * Indicates whether the magnetometer  has new measurement data ready.
+     * 
+     * @return True if there is new magnetometer data available; false otherwise.
+     */
+    bool magnetometerDataReady() {return true;}
+    
+    /**
+     * Get last raw Accelerometer values as a IMUData struct containing values in x, y and z.
+     *
+     * @param[in] accelerationValues  Pointer to IMUData struct.
+     */
+    const void getAccelerationValues(IMUData* accelerationValues) {}
+
+    /**
+     * Get last raw Gyroscope values as a IMUData struct containing values in x, y and z.
+     *
+     * @param[in] turnRates  Pointer to IMUData struct.
```

Comment:
Unit?

Reply:
Added an explanation why the value is unitless.